server/channels/app/metrics.go

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"net/http"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/shared/request"
+)
+
+func (a *App) RegisterPerformanceReport(rctx request.CTX, report *model.PerformanceReport) *model.AppError {
+	if a.Metrics() == nil {
+		return nil
+	}
+
+	for _, c := range report.Counters {
+		switch c.Metric {
+		case model.ClientMetricChannelVisited:
+			a.Metrics().IncrementClientChannelVisited(report.Labels["platform"], report.Labels["agent"], float64(c.Value))
```

Comment:
Were you planning on bucketing these values on the server still? I was thinking that making the server responsible for that reduces the chance that a rogue client tries to cause problems by making to many timeserieses for these metrics

Reply:
Sure, these labels should be filtered out on server side. I'll add that logic.